File: Core/src/main/java/dev/lynith/core/bridge/gui/MCScreen.java
package dev.lynith.core.bridge.gui;

public class MCScreen {

    public void render(int mouseX, int mouseY, float delta) {}
    public void init() {}

    public void closed() {}
    public void keyPressed(char typedChar, int keyCode) {}
    public void mouseClicked(int mouseX, int mouseY, int button) {}
    public void mouseReleased(int mouseX, int mouseY, int button) {}
    public void mouseDragged(int mouseX, int mouseY, int button, float time) {}

    public boolean shouldPauseGame() { return false; }

    public void resized() {}
}

File: Core/src/main/kotlin/dev/lynith/core/bridge/gui/MCScreen.kt
package dev.lynith.core.bridge.gui

open class MCScreen {

    open fun render(mouseX: Int, mouseY: Int, delta: Float) {}
    open fun init() {}

    open fun keyPressed(typedChar: Char, keyCode: Int) {}
    open fun mouseClicked(mouseX: Int, mouseY: Int, button: Int) {}
    open fun mouseReleased(mouseX: Int, mouseY: Int, button: Int) {}
    open fun mouseDragged(mouseX: Int, mouseY: Int, button: Int, time: Float) {}

    open fun closed() {}
    open fun resized() {}

    open fun shouldPauseGame(): Boolean {
        return false
    }
}
